app/assets/javascripts/discourse/app/models/topic-tracking-state.js

```diff
@@ -610,14 +621,38 @@ const TopicTrackingState = EmberObject.extend({
   },
 
   loadStates(data) {
-    (data || []).forEach((topic) => {
-      this.modifyState(topic, topic);
+    if (!data || data.length === 0) {
+      return;
+    }
+
+    const modified = data.every((topic) => {
+      return this._setState({ topic, data: topic, skipAfterStateChange: true });
     });
+
+    if (modified) {
+      this._afterStateChange();
+    }
+  },
+
+  _setState({ topic, data, skipAfterStateChange } = {}) {
```

Comment:
Do we ever call/want to call `_setState` with no args?

```suggestion
  _setState({ topic, data, skipAfterStateChange }) {
```